### PR TITLE
In HostLegacyRouting=true, avoid rp_filter drop via rule/route config

### PR DESCRIPTION
**Pull request by lyxly**

### CODE REVIEW

**Comment by lyxly** [comment]
cilium-cni side triger the primary nic default route rule. Best way is adding in CiliumNode operator, optimize it next step.

File: pkg/datapath/linux/routing/routing.go [hunk 1]
```diff
@@ -135,6 +135,153 @@ func (info *RoutingInfo) Configure(ip net.IP, mtu int, compat bool) error {
 	return nil
 }
 
+// In HostLegacyRouting, avoid rp_filter check
+func (info *RoutingInfo) ConfigureForLegacy(ip net.IP, mtu int, compat bool) error {
+	log.Errorf("############## RoutingInfo is %+v", info)
```
Comment (chestack) [inline pkg/datapath/linux/routing/routing.go:140]: please change to log.Debug()

File: pkg/datapath/linux/routing/routing.go [hunk 2]
```diff
@@ -135,6 +135,153 @@ func (info *RoutingInfo) Configure(ip net.IP, mtu int, compat bool) error {
...
+	if err != nil {
+		return fmt.Errorf("unable to find ifindex for interface MAC: %s", err)
+	}
+
+	ipWithMask := net.IPNet{
+		IP:   ip,
+		Mask: net.CIDRMask(32, 32),
+	}
+
+	log.Errorf("############## ip is %+v, ifindex is %d", ip, ifindex)
```
Comment (chestack) [inline pkg/datapath/linux/routing/routing.go:159]: please change to log.Debug()

File: pkg/datapath/linux/routing/routing.go [hunk 3]
```diff
@@ -135,6 +135,153 @@ func (info *RoutingInfo) Configure(ip net.IP, mtu int, compat bool) error {
...
+	var egressPriority, tableID int
+	if compat {
+		egressPriority = linux_defaults.RulePriorityEgress
+		tableID = ifindex
+	} else {
+		egressPriority = linux_defaults.RulePriorityEgressv2
+		tableID = computeTableIDFromIfaceNumber(info.InterfaceNumber)
+	}
+
+	log.Errorf("############## tableID is %d, priority is %d, From is %+v", tableID, egressPriority, ipWithMask)
```
Comment (chestack) [inline pkg/datapath/linux/routing/routing.go:180]: please change to log.Infof()

File: pkg/datapath/linux/routing/routing.go [hunk 4]
```diff
@@ -135,6 +135,153 @@ func (info *RoutingInfo) Configure(ip net.IP, mtu int, compat bool) error {
...
+		}
+	}
+	// Get control plane ip/network for cidr filtering
+	primaryAddrs, _ := netlink.AddrList(primaryLink, netlink.FAMILY_V4)
+	primaryAddr := primaryAddrs[0]
+	primaryAddrIp := primaryAddr.IP
+	primaryNetwork := primaryAddrIp.Mask(primaryAddr.Mask)
+	//log.Errorf("#### primary interface ip:%v", primaryAddrIp)     // 192.168.0.153
+	//log.Errorf("#### primary interface mask:%v", primaryAddr.Mask) // ffffff00
+	//log.Errorf("#### primary interface mask:%v", primaryNetwork) // 192.168.0.0
```
Comment (chestack) [inline pkg/datapath/linux/routing/routing.go:226]: please remove these using commented out message

File: pkg/datapath/linux/routing/routing.go [hunk 5]
```diff
@@ -135,6 +135,153 @@ func (info *RoutingInfo) Configure(ip net.IP, mtu int, compat bool) error {
...
+		LinkIndex: primaryLink.Attrs().Index,
+		Dst:       &net.IPNet{IP: net.IPv4zero, Mask: net.CIDRMask(0, 32)},
+		Table:     tableID,
+		//Gw:       info.IPv4Gateway,
+		Gw:       primaryDefaultRoute.Gw,
+		Protocol: linux_defaults.RTProto,
+	}); err != nil {
+		return fmt.Errorf("unable to add L2 nexthop route: %s", err)
+	}
+	//log.Errorf("#####primaryDefaultRoute.LinkIndex##%+v, ", primaryDefaultRoute.LinkIndex)
```
Comment (chestack) [inline pkg/datapath/linux/routing/routing.go:264]: please remove these lines

**Pull request closed without merging**